### PR TITLE
ROX-18192: Fix sensor reconcile tests

### PR DESCRIPTION
**Pull request by vikin91**
## Description

The assertion checking for number of deletions had an upper limit of 4, while we were observing 5 deletions. The 5th one was the scanner pod being scaled down.

In this PR, we changed the assertion, so that the names of pods are taken into account instead of counting totals.

## Checklist
- [x] Investigated and inspected CI test results

### N/A
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

## Testing Performed
- [x] CI only

### CODE REVIEW

File: sensor/tests/compliance/compliance_test.go
Comment (fredrb) [inline sensor/tests/compliance/compliance_test.go:1]: Is this supposed to be here? I see that you branched from a compliance integration test branch. But I wonder if this was by mistake?

Comment (lvalerom) [reply sensor/tests/compliance/compliance_test.go:1]: +1 I think it was a mistake

Comment (vikin91) [reply sensor/tests/compliance/compliance_test.go:1]: Thanks! It should not be here, let me fix that

**Review by lvalerom — APPROVED** [review]
The test runs in the compatibility and e2e steps look green 🎉

**Comment by vikin91** [comment]
Let's ship that then and see what happens :)

**Pull request merged**